### PR TITLE
tikv: drop store's regions when resolve store with tombstone status

### LINKED ISSUE #22907: Successive StoreNotMatch errors after add new store with tombstone store's IP
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

0. start a region with 3 store(s1, s2, s3), and s1 is leader
1. make s1 as tombstone
2. add new s4 with s3's ip and make s4 as region leader
3. invoke query to hit region leader

### 2. What did you expect to see? (Required)

some storeNotMatch and recover quickly

### 3. What did you see instead (Required)

Successive StoreNotMatch errors after replace tombstone store with new store id and timeout for user query.

### 4. What is your TiDB version? (Required)

```
Release Version: None
Edition: Community
Git Commit Hash: None
Git Branch: None
UTC Build Time: None
GoVersion: go1.15
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by lysu**
Signed-off-by: lysu <sulifx@gmail.com>

### What problem does this PR solve?

Issue Number: close #22907

Problem Summary:

old store id can get old tombstone store, so never have chance to refresh region peer to point to new store id

so request will block forever in old store with EpochNotMatchError

### What is changed and how it works?

What's Changed:

drop region cache when resolve store but it's tombstone status

How it Works:

after it, retry will re-fetch new region info from PD which point to new store id..

so continue request will got to new store and never block

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch 4.0/3.0

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- n/a

### Release note 

- Fix request block forever when replace a tombstone store with the store has same IP addr

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/22909)

### CODE REVIEW

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -1516,7 +1516,7 @@ func (s *Store) reResolve(c *RegionCache) {
 		// we cannot do backoff in reResolve loop but try check other store and wait tick.
 		return
 	}
-	if store == nil {
+	if store == nil || store.State == metapb.StoreState_Tombstone {
```
Comment (lysu) [inline store/tikv/region_cache.go:1519]: @HunDunDM question, do we need drop region when state be `StoreState_Offline`?

**Comment by coocood** [comment]
LGTM

File: store/tikv/region_cache.go (hunk 1 above)
Comment (youjiali1995) [reply store/tikv/region_cache.go:1519]: `store == nil` means the store is a tombstone. If the store doesn't exist, `pdClient.GetStore()` returns an error: https://github.com/tikv/pd/blob/bc63de897afa69cfcc5828f45431fe3999fed4b3/client/client.go#L1159-L1168

**Pull request merged**